core/src/main/java/org/axonframework/eventsourcing/EventSourcingRepository.java

```diff
@@ -204,7 +204,8 @@ public EventSourcingRepository(AggregateFactory<T> aggregateFactory, EventStore
      */
     @Override
     protected EventSourcedAggregate<T> doLoadWithLock(String aggregateIdentifier, Long expectedVersion) {
```

Comment:
Would you mind introducing a test which calls the  chance `EventSourcingRepository#doLoadWithLock(String, Long)` function which ensures the filtering is performed?